lib/awful/layout/suit/tile.lua.in

```diff
@@ -156,26 +155,28 @@ local function tile_group(cls, wa, orientation, fact, group)
     size = math.min(size, available)
 
     local coord = wa[y]
-    local geom = {}
     local used_size = 0
     local unused = wa[height]
     for c = group.first,group.last do
         local i = c - group.first +1
-        geom[width] = size - cls[c].border_width * 2
-        geom[height] = math.floor(unused * fact[i] / total_fact) - cls[c].border_width * 2
+        local geom = {}
+        geom[width] = size
+        geom[height] = math.floor(unused * fact[i] / total_fact)
         geom[x] = group.coord
         geom[y] = coord
-        geom = cls[c]:geometry(geom)
```

Comment:
Not in every case, but you can look at the `size_hints` property of clients 

https://awesome.naquadah.org/doc/api/modules/client.html

I don't remember all the details of what this property provide, you will have to try it experimentally.